intellij/src/nl/tudelft/watchdog/intellij/logic/ui/listeners/EditorWindowListener.java

```diff
@@ -1,53 +1,76 @@
 package nl.tudelft.watchdog.intellij.logic.ui.listeners;
 
+import com.intellij.openapi.Disposable;
 import com.intellij.openapi.editor.Editor;
 import com.intellij.openapi.editor.event.EditorFactoryEvent;
 import com.intellij.openapi.editor.event.EditorFactoryListener;
+import com.intellij.openapi.editor.impl.DocumentMarkupModel;
+import com.intellij.openapi.editor.impl.MarkupModelImpl;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Disposer;
+import nl.tudelft.watchdog.core.logic.event.TrackingEventManager;
+import org.jetbrains.annotations.NotNull;
 
 import java.util.HashMap;
 import java.util.Map;
 
-public class EditorWindowListener implements EditorFactoryListener {
+public class EditorWindowListener implements EditorFactoryListener, Disposable {
 
     private EditorFocusListener focusListener;
 
-    private Map<Editor, EditorListener> editorListenerMap = new HashMap<Editor, EditorListener>();
+    /**
+     * Every editor that is created is associated with a tree of disposables.
+     * The parent Disposable is registered with each associated listener to
+     * events of the editor. By using Disposable for this structure, we can
```

Comment:
Link to Disposable?

Reply:
It is already in the type information, but added it nonetheless.